host/src/lib/HelperFunctions.jsx

```diff
@@ -347,39 +343,45 @@ export const checkEqualityWithOtherAnswers = (rawAnswer, normValue, normType, pr
  * @returns {IResponse[]}
  */
 export const buildShortAnswerResponses = (prevShortAnswer, choices, newAnswer, newAnswerTeamName, teamId) => {
+  let correctAnswer = choices.find(choice => choice.isAnswer).text;
+  const correctAnswerRegEx = new RegExp(`\\b${correctAnswer.toLowerCase()}\\b`, 'g');
   if (prevShortAnswer.length === 0) {
-    const correctAnswer = choices.find(choice => choice.isAnswer).text;
     const correctAnswerType = determineAnswerType(correctAnswer);
-
+    if (correctAnswerType === AnswerType.NUMBER) {
+      correctAnswer = Number(correctAnswer);
+    } else {
+      correctAnswer = correctAnswer.toLowerCase();
+    }
+   
     prevShortAnswer.push({
       value: correctAnswer,
       isCorrect: true,
       isSelectedMistake: false,
-      normAnswer: [{
-        value: correctAnswer,
-        type: correctAnswerType,
-      }],
+      normAnswer: {
+        [correctAnswerType]: [correctAnswer]
+      },
       count: 0,
       teams: [],
     });
   }
   const rawAnswer = newAnswer.answerContent.rawAnswer;
   let isExistingAnswer = false;
-  outerloop:
-  // for each normalized answer in the newly submitted answer
-  for (let i = 0; i < newAnswer.answerContent.normAnswer.length; i++) {
-    const answer = newAnswer.answerContent.normAnswer[i];
-    // for each answer in the previous short answer array 
-    for (let y = 0; y < prevShortAnswer.length; y++) {
-      if (checkEqualityWithOtherAnswers(rawAnswer, answer.value, answer.type, prevShortAnswer[y])) {
-        isExistingAnswer = true;
-        prevShortAnswer[y].count += 1;
-        prevShortAnswer[y].teams.push({name: newAnswerTeamName, id: teamId, confidence: newAnswer.confidenceLevel});
-        break outerloop;
-      }
-    };
-  };
-      
+
+  // for each answer type in the newly submitted answer
```

Comment:
for each type of answer, will call the equality function using the corresponding type of the previous answer.